morphem/Feature_Extraction.py

```diff
@@ -0,0 +1,142 @@
+import torch
+from torch.utils.data import DataLoader
+from torch import nn
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from skimage import io
+import skimage
+import importlib
+from tqdm import tqdm
+
+from torchvision.models import resnet18, ResNet18_Weights
+from timm import create_model
+import torch.nn.functional as F
+
+import argparse
+
+
+
+import folded_dataset
+# reload(folded_dataset)
+
+
+def configure_dataset(root_dir, dataset_name):
+    df_path = f'{root_dir}/{dataset_name}/enriched_meta.csv'
+    df = pd.read_csv(df_path)
+    dataset = folded_dataset.SingleCellDataset(csv_file=df_path, root_dir=root_dir, target_labels='train_test_split')
+
+    return dataset
+
+
+
+class ConvNextClass():
+    def convnext_model(gpu):
+        device = torch.device(f"cuda:{gpu}" if torch.cuda.is_available() else "cpu")
+
+      
+        feature_file = 'pretrained_convnext_channel_replicate.npy'
+     
+
+        pretrained = True
+        model = create_model("convnext_tiny.fb_in22k", pretrained=pretrained).to(device)
+        feature_extractor = nn.Sequential(
+                            model.stem,
+                            model.stages[0],
+                            model.stages[1],
+                            model.stages[2].downsample,
+                            *[model.stages[2].blocks[i] for i in range(9)],
+                            model.stages[3].downsample,
+                            *[model.stages[3].blocks[i] for i in range(3)],
+                        )
+
+        model_check = "convnext"
+        return feature_extractor, device, feature_file
+
+
+    def forward(x: torch.Tensor, gpu: int) -> torch.Tensor:
+        feature_extractor, _,  = ConvNextClass.convnext_model(gpu)
+        x = feature_extractor(x)
```

Comment:
Here, the feature extractor should come from the self object:
`x = self.feature_extractor(x)`